<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DateLife Workflow • datelife</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DateLife Workflow">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">datelife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/datelife.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/datelife_examples.html">DateLife Examples</a>
    </li>
    <li>
      <a href="../articles/datelife_workflow.html">DateLife Workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/phylotastic/datelife">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>DateLife Workflow</h1>
                        <h4 class="author">Luna L. Sanchez Reyes</h4>
            
            <h4 class="date">2019-12-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/phylotastic/datelife/blob/master/vignettes/datelife_workflow.Rmd"><code>vignettes/datelife_workflow.Rmd</code></a></small>
      <div class="hidden name"><code>datelife_workflow.Rmd</code></div>

    </div>

    
    
<p>DateLife is a service for searching and processing information on ages of any number of taxa <!--or group of taxa-->of interest, across chronograms available in public data repositories coming from published peer reviewed studies. It can also generate new taxon age information<!--by using expert available phylogenies and scaling them
to absolute time using information on ages obtained from published chronograms,--> by linking several external services and tested algorithms.</p>
<p>It only requires a set of taxon names as input, in the form of a comma separated listing or vector, or of a phylogeny with taxon names on the tips. Taxon names can correspond to binomial species names or clades <!--common names (will we implement it?)-->. When taxon names are clades, DateLife pulls all accepted species names within the clade (up to OToL’s limit of _____ species) from OToL’s reference taxonomy using a service of rphylotastic R package<!--citation-->. Names belonging to subspecies or any other infraspecific category are treated as species. DateLife can process input names with the taxon name resolution service (TNRS), <!--which "corrects misspelled names and authorities, standardizes variant spellings,
and converts nomenclatural synonyms to accepted names" (Boyle et al. 2013)--> which corrects misspelled names or typos, and standardizes variation in spelling and synonyms <span class="citation">[@Boyle2013]</span>, increasing the probability to correctly find the queried taxa in the chronogram database. DateLife uses TNRS to compare names against OToL’s reference taxonomy using a service from the R package rotl <span class="citation">[@Michonneau2016]</span>. <!--we'll implement the option to use other reference taxonomies in the future --></p>
<p>DateLife’s main function searches taxon names across the chronogram database specified by the user. At the moment, it queries chronograms from OToL <span class="citation">[@Hinchliff2015]</span> <!-- it cannot query chronograms from TreeBASE -there is an earlier reference from
1994 but it does not exist in its journal database: Sanderson MJ, Donoghue MJ, Piel
WH, Eriksson T: TreeBASE: a prototype database of phylogenetic analyses and an interactive
tool for browsing the phylogeny of life. Am J Bot. 1994, 81 (6): 183--> repository. DateLife identifies chronograms having at least two taxon names, and subset them to contain only the taxa of interest. It then stores taxon age information from each chronogram individually as a patristic matrix, named with the citation of the original study. This format allows a rapid summary in a number of different ways, including: 1) citations of the original studies containing the subset chronograms, 2) a list of mrca ages of subset chronograms, 3) a list of complete subset chronograms in newick or phylo format, 4) a table containing all information retrieved in html or R’s data frame format, or 5) a single chronogram summarized from subset chronograms using the Super Distance Matrix (SDM) supertree construction approach <span class="citation">[@Criscuolo2006]</span> or using the median of branch lengths. <!--(how to better explain the latter?? Also, I just discovered the median gives non ultrametric trees...).--></p>
<p>DateLife also stores information on input taxon presence/absence across subset chronograms. Users can choose to add ages of missing taxa to subset chronograms in different ways, depending on the amount of knowledge they want to input or how much they want to be involved in the steps of the addition process. If users have no access to biological information (i.e., a character, DNA or protein matrix), missing taxa can be added to any chronogram simply at random, or by following taxonomic or phylogenetic knowledge from expert sources. There are a wide number of open reference taxonomies available, such as the Catalogue of Life <span class="citation">[@Roskov2017]</span> or the NCBI taxonomy database <span class="citation">[@Federhen2012]</span>. Expert phylogenies (with or without branch lengths) to be used as topological constraint (backbone) can also be obtained from a number of public repositories, such as OToL <span class="citation">[@Hinchliff2015]</span>, TreeBASE <span class="citation">[@Piel2002]</span> and Dryad (<a href="https://www.datadryad.org//" class="uri">https://www.datadryad.org//</a>). At the moment, DateLife only uses OToL’s synthetic tree and reference taxonomy as expert knowledge to automatically add missing taxa to chronograms. Alternatively, users can input a reference taxonomy or topological constraint of their choosing or making. <!-- at some point we will implement the possibility for users to choose
different reference taxonomies automatically--> If OToL’s synthetic tree is not satisfactorily resolved for the taxa of interest, DateLife can construct a sequence data matrix from DNA markers available from the Barcode of Life Database (BOLD; <span class="citation">@Ratnasingham2007</span>), to attempt to further resolve polytomies. It will follow OToL’s synthetic tree as backbone. <!--This molecular tree can then be dated with chronosMPL from the ape package, but is it really needed?? To estimate node ages chronosMPL uses mean path length method from Britton et al. 2002--> <!--It can rapidly construct trees with branch lengths from
a set of lineages with sequence data from the Barcode Of Life Database (BOLD, <http://www.boldsystems.org/>), using the synthetic OToL (<https://tree.opentreeoflife.org>) as backbone, and to obtain divergence dates.
It allows direct comparison of dates obtained with different markers
available in BOLD (in plants and fungi in particular).--> To use information from a topological constraint, DateLife calls the congruification method described in <span class="citation">[@Eastman2013]</span> to find shared nodes between trees (congruent nodes). It then fixes their ages, and add ages to remaining nodes with a dating method that can be specified by the user. If users have access to biological data, they can input a tree with branch lengths proportional to relative substitution rates as topological constraint. In this case, age data from congruent nodes will be used as calibration points. Age data from several chronograms can be combined and congruified to be used as calibration points in a single analysis.</p>
<p>Several dating methods are implemented in DateLife. Branch Length Adjuster (BLADJ) is a simple algorithm to distribute ages of undated nodes evenly, which minimizes age variance in the chronogram <span class="citation">[@Webb2008]</span>. DateLife implements BLADJ from the development R version of phylocom’s R package <span class="citation">[@Webb2008]</span>, phylocomr (<a href="https://github.com/ropensci/phylocomr" class="uri">https://github.com/ropensci/phylocomr</a>). It can only be used when there is a topological constraint with no branch lengths. PATHd8 is a non-clock, rate-smoothing method <span class="citation">[@Britton2007]</span> to date trees. It is also called through R. <!--It could be used with a reference taxonomy, if we provide a tree with politomies--> treePL, is a semi-parametric, rate-smoothing, penalized likelihood dating method <span class="citation">[@Smith2012]</span>. It is called through R. MrBayes program <span class="citation">[@Huelsenbeck2001; @Ronquist2003]</span> can be used when adding taxa at random, following a reference taxonomy or a topological constraint. It draws ages from a pure birth model, as implemented by Jetz and collaborators <span class="citation">[-@Jetz2012]</span>. DateLife calls MrBayes trough an R function.</p>
<p>DateLife can also correct negative branch lengths in several ways.</p>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian O'Meara, Luna L. Sanchez-Reyes, Jonathan Eastman, Tracy Heath, April Wright, Klaus Schliep, Scott Chamberlain, Peter Midford, Luke Harmon, Joseph Brown, Matt Pennell, Mike Alfaro.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
