<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Core function to input a vector of species, newick string, or phylo object to get a chronogram or dates back. — datelife_search • datelife</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Core function to input a vector of species, newick string, or phylo object to get a chronogram or dates back. — datelife_search" />
<meta property="og:description" content="Core function to input a vector of species, newick string, or phylo object to get a chronogram or dates back." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">datelife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/datelife.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/datelife_examples.html">DateLife Examples</a>
    </li>
    <li>
      <a href="../articles/datelife_workflow.html">DateLife Workflow</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/phylotastic/datelife">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Core function to input a vector of species, newick string, or phylo object to get a chronogram or dates back.</h1>
    <small class="dont-index">Source: <a href='https://github.com/phylotastic/datelife/blob/master/R/datelife_search.R'><code>R/datelife_search.R</code></a></small>
    <div class="hidden name"><code>datelife_search.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Core function to input a vector of species, newick string, or phylo object to get a chronogram or dates back.</p>
    </div>

    <pre class="usage"><span class='fu'>datelife_search</span>(
  <span class='kw'>input</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"Rhea americana"</span>, <span class='st'>"Pterocnemia pennata"</span>, <span class='st'>"Struthio camelus"</span>),
  <span class='kw'>summary_format</span> <span class='kw'>=</span> <span class='st'>"phylo_all"</span>,
  <span class='kw'>partial</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>use_tnrs</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>approximate_match</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>update_cache</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>cache</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/get.html'>get</a></span>(<span class='st'>"opentree_chronograms"</span>),
  <span class='kw'>summary_print</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"citations"</span>, <span class='st'>"taxa"</span>),
  <span class='kw'>taxon_summary</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"none"</span>, <span class='st'>"summary"</span>, <span class='st'>"matrix"</span>),
  <span class='kw'>get_spp_from_taxon</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>criterion</span> <span class='kw'>=</span> <span class='st'>"taxa"</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>input</th>
      <td><p>Target taxa names as a character vector, a newick character string, or a phylo object.</p></td>
    </tr>
    <tr>
      <th>summary_format</th>
      <td><p>The desired output format for target chronograms (chronograms of target taxa). See details.</p></td>
    </tr>
    <tr>
      <th>partial</th>
      <td><p>If TRUE, use source chronograms even if they only match some of the desired taxa</p></td>
    </tr>
    <tr>
      <th>use_tnrs</th>
      <td><p>If TRUE, use OpenTree's services to resolve names. This can dramatically improve the chance of matches, but also take much longer.</p></td>
    </tr>
    <tr>
      <th>approximate_match</th>
      <td><p>If TRUE, use a slower TNRS to correct misspellings, increasing the chance of matches (including false matches).</p></td>
    </tr>
    <tr>
      <th>update_cache</th>
      <td><p>default to FALSE</p></td>
    </tr>
    <tr>
      <th>cache</th>
      <td><p>The cached set of chronograms and other info from data(opentree_chronograms).</p></td>
    </tr>
    <tr>
      <th>summary_print</th>
      <td><p>A character vector specifying type of summary information to be printed: "citations" for the references of chronograms from cache where target taxa are found, "taxa" for a summary of the number of chronograms where each target taxon is found, or "none" if nothing should be printed. Default to display both c("citations", "taxa").</p></td>
    </tr>
    <tr>
      <th>taxon_summary</th>
      <td><p>A character vector specifying if data on target taxa missing in source chronograms should be added to the output as a "summary" or as a presence/absence "matrix". Default to "none", no information on taxon_summary added to the output.</p></td>
    </tr>
    <tr>
      <th>get_spp_from_taxon</th>
      <td><p>boolean vector, default to FALSE. If TRUE, will get all species names from taxon names given in input. Must have same length as input. If input is a newick string , with some clades it will be converted to phylo object phy, and the order of get_spp_from_taxon will match phy$tip.label.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Boolean. If TRUE, it gives printed updates to the user.</p></td>
    </tr>
    <tr>
      <th>criterion</th>
      <td><p>Whether to get the grove with the most trees or the most taxa</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Available output formats are:</p>
<p>citations: A character vector of references where chronograms with some or all of the target taxa are published (source chronograms).</p>
<p>mrca: A named numeric vector of most recent common ancestor (mrca) ages of target taxa defined in input, obtained from the source chronograms. Names of mrca vector are equal to citations.</p>
<p>newick_all: A named character vector of newick strings corresponding to target chronograms derived from source chronograms. Names of newick_all vector are equal to citations.</p>
<p>newick_sdm: Only if multiple source chronograms are available. A character vector with a single newick string corresponding to a target chronogram obtained with SDM supertree method (Criscuolo et al. 2006).</p>
<p>newick_median: Only if multiple source chronograms are available. A character vector with a single newick string corresponding to a target chronogram from the median of all source chronograms.</p>
<p>phylo_sdm: Only if multiple source chronograms are available. A phylo object with a single target chronogram obtained with SDM supertree method (Criscuolo et al. 2006).</p>
<p>phylo_median: Only if multiple source chronograms are available. A phylo object with a single target chronogram obtained from source chronograms with median method.</p>
<p>phylo_all: A named list of phylo objects corresponding to each target chronogram obtained from available source chronograms. Names of phylo_all list correspond to citations.</p>
<p>phylo_biggest: The chronogram with the most taxa. In the case of a tie, the chronogram with clade age closest to the median age of the equally large trees is returned.</p>
<p>html: A character vector with an html string that can be saved and then opened in any web browser. It contains a 4 column table with data on target taxa: mrca, number of taxa, citations of source chronogram and newick target chronogram.</p>
<p>data_frame A data frame with data on target taxa: mrca, number of taxa, citations of source chronograms and newick string.</p>
<p>For approaches that return a single synthetic tree, it is important that the trees leading to it form a grove (roughly, a sufficiently overlapping set of taxa between trees: see Ané et al. 2005, 10.1007/s00026-009-0017-x). In the rare case of multiple groves, should we take the one with the most trees or the most taxa?</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># obtain median ages from a set of source chronograms in newick format:</span>
<span class='no'>ages</span> <span class='kw'>&lt;-</span> <span class='fu'>datelife_search</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"Rhea americana"</span>, <span class='st'>"Pterocnemia pennata"</span>, <span class='st'>"Struthio camelus"</span>,
                <span class='st'>"Mus musculus"</span>), <span class='kw'>summary_format</span><span class='kw'>=</span><span class='st'>"newick_median"</span>)</div><div class='output co'>#&gt; <span class='message'>Trying with overlap = 2</span></div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</div><div class='output co'>#&gt; <span class='message'>Source chronograms from:</span></div><div class='output co'>#&gt; <span class='message'>1: Grealy, Alicia, Matthew Phillips, Gifford Miller, M. Thomas P. Gilbert, Jean-Marie Rouillard, David Lambert, Michael Bunce, James Haile. 2017. Eggshell palaeogenomics: Palaeognath evolutionary history revealed through ancient nuclear and mitochondrial DNA from Madagascan elephant bird (Aepyornis sp.) eggshell. Molecular Phylogenetics and Evolution 109: 151-163</span></div><div class='output co'>#&gt; <span class='message'>2: Hedges, S. Blair, Julie Marin, Michael Suleski, Madeline Paymer, Sudhir Kumar. 2015. Tree of life reveals clock-like speciation and diversification. Molecular Biology and Evolution 32 (4): 835-845</span></div><div class='output co'>#&gt; <span class='message'>3: Hedges, S. Blair, Julie Marin, Michael Suleski, Madeline Paymer, Sudhir Kumar. 2015. Tree of life reveals clock-like speciation and diversification. Molecular Biology and Evolution 32 (4): 835-845</span></div><div class='output co'>#&gt; <span class='message'>4: Prum, Richard O., Jacob S. Berv, Alex Dornburg, Daniel J. Field, Jeffrey P. Townsend, Emily Moriarty Lemmon, Alan R. Lemmon. 2015. A comprehensive phylogeny of birds (Aves) using targeted next-generation DNA sequencing. Nature 526, (7574): 569-573</span></div><div class='output co'>#&gt; <span class='message'>5: Prum, Richard O., Jacob S. Berv, Alex Dornburg, Daniel J. Field, Jeffrey P. Townsend, Emily Moriarty Lemmon, Alan R. Lemmon. 2015. A comprehensive phylogeny of birds (Aves) using targeted next-generation DNA sequencing. Nature 526, (7574): 569-573</span></div><div class='output co'>#&gt; <span class='message'>6: Claramunt, Santiago, Joel Cracraft. 2015. A new time tree reveals Earth historys imprint on the evolution of modern birds. Science Advances 1 (11): e1501005-e1501005</span></div><div class='output co'>#&gt; <span class='message'>7: Jetz, W., G. H. Thomas, J. B. Joy, K. Hartmann, A. O. Mooers. 2012. The global diversity of birds in space and time. Nature 491 (7424): 444-448</span></div><div class='output co'>#&gt; <span class='message'>8: Jetz, W., G. H. Thomas, J. B. Joy, K. Hartmann, A. O. Mooers. 2012. The global diversity of birds in space and time. Nature 491 (7424): 444-448</span></div><div class='output co'>#&gt; <span class='message'>9: Mitchell, K. J., B. Llamas, J. Soubrier, N. J. Rawlence, T. H. Worthy, J. Wood, M. S. Y. Lee, A. Cooper. 2014. Ancient DNA reveals elephant birds and kiwi are sister taxa and clarifies ratite bird evolution. Science 344 (6186): 898-900.</span></div><div class='output co'>#&gt; <span class='message'>Input taxa presence across source chronograms:</span></div><div class='output co'>#&gt; <span class='message'>                taxon chronograms</span>
#&gt; <span class='message'>1 Pterocnemia pennata         6/9</span>
#&gt; <span class='message'>2      Rhea americana         9/9</span>
#&gt; <span class='message'>3    Struthio camelus         9/9</span>
#&gt; <span class='message'>4        Mus musculus         1/9</span></div><div class='output co'>#&gt; <span class='message'>Input taxa completely absent from source chronograms:</span></div><div class='output co'>#&gt; <span class='message'>  taxon</span>
#&gt; <span class='message'>1  None</span></div><div class='input'><span class='co'># save the tree in newick format</span>
<span class='fu'><a href='https://rdrr.io/r/base/write.html'>write</a></span>(<span class='no'>ages</span>, <span class='kw'>file</span><span class='kw'>=</span><span class='st'>"some.bird.ages.txt"</span>)

<span class='co'># obtain median ages from a set of source chronograms in phylo format</span>
<span class='co'># will produce same tree as above but in r phylo format:</span>
<span class='no'>ages.again</span> <span class='kw'>&lt;-</span> <span class='fu'>datelife_search</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"Rhea americana"</span>, <span class='st'>"Pterocnemia pennata"</span>, <span class='st'>"Struthio camelus"</span>,
                <span class='st'>"Mus musculus"</span>), <span class='kw'>summary_format</span><span class='kw'>=</span><span class='st'>"phylo_median"</span>)</div><div class='output co'>#&gt; <span class='message'>Trying with overlap = 2</span></div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</div><div class='output co'>#&gt; <span class='message'>Source chronograms from:</span></div><div class='output co'>#&gt; <span class='message'>1: Grealy, Alicia, Matthew Phillips, Gifford Miller, M. Thomas P. Gilbert, Jean-Marie Rouillard, David Lambert, Michael Bunce, James Haile. 2017. Eggshell palaeogenomics: Palaeognath evolutionary history revealed through ancient nuclear and mitochondrial DNA from Madagascan elephant bird (Aepyornis sp.) eggshell. Molecular Phylogenetics and Evolution 109: 151-163</span></div><div class='output co'>#&gt; <span class='message'>2: Hedges, S. Blair, Julie Marin, Michael Suleski, Madeline Paymer, Sudhir Kumar. 2015. Tree of life reveals clock-like speciation and diversification. Molecular Biology and Evolution 32 (4): 835-845</span></div><div class='output co'>#&gt; <span class='message'>3: Hedges, S. Blair, Julie Marin, Michael Suleski, Madeline Paymer, Sudhir Kumar. 2015. Tree of life reveals clock-like speciation and diversification. Molecular Biology and Evolution 32 (4): 835-845</span></div><div class='output co'>#&gt; <span class='message'>4: Prum, Richard O., Jacob S. Berv, Alex Dornburg, Daniel J. Field, Jeffrey P. Townsend, Emily Moriarty Lemmon, Alan R. Lemmon. 2015. A comprehensive phylogeny of birds (Aves) using targeted next-generation DNA sequencing. Nature 526, (7574): 569-573</span></div><div class='output co'>#&gt; <span class='message'>5: Prum, Richard O., Jacob S. Berv, Alex Dornburg, Daniel J. Field, Jeffrey P. Townsend, Emily Moriarty Lemmon, Alan R. Lemmon. 2015. A comprehensive phylogeny of birds (Aves) using targeted next-generation DNA sequencing. Nature 526, (7574): 569-573</span></div><div class='output co'>#&gt; <span class='message'>6: Claramunt, Santiago, Joel Cracraft. 2015. A new time tree reveals Earth historys imprint on the evolution of modern birds. Science Advances 1 (11): e1501005-e1501005</span></div><div class='output co'>#&gt; <span class='message'>7: Jetz, W., G. H. Thomas, J. B. Joy, K. Hartmann, A. O. Mooers. 2012. The global diversity of birds in space and time. Nature 491 (7424): 444-448</span></div><div class='output co'>#&gt; <span class='message'>8: Jetz, W., G. H. Thomas, J. B. Joy, K. Hartmann, A. O. Mooers. 2012. The global diversity of birds in space and time. Nature 491 (7424): 444-448</span></div><div class='output co'>#&gt; <span class='message'>9: Mitchell, K. J., B. Llamas, J. Soubrier, N. J. Rawlence, T. H. Worthy, J. Wood, M. S. Y. Lee, A. Cooper. 2014. Ancient DNA reveals elephant birds and kiwi are sister taxa and clarifies ratite bird evolution. Science 344 (6186): 898-900.</span></div><div class='output co'>#&gt; <span class='message'>Input taxa presence across source chronograms:</span></div><div class='output co'>#&gt; <span class='message'>                taxon chronograms</span>
#&gt; <span class='message'>1 Pterocnemia pennata         6/9</span>
#&gt; <span class='message'>2      Rhea americana         9/9</span>
#&gt; <span class='message'>3    Struthio camelus         9/9</span>
#&gt; <span class='message'>4        Mus musculus         1/9</span></div><div class='output co'>#&gt; <span class='message'>Input taxa completely absent from source chronograms:</span></div><div class='output co'>#&gt; <span class='message'>  taxon</span>
#&gt; <span class='message'>1  None</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='no'>ages.again</span>)</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='no'>ape</span>)
<span class='kw pkg'>ape</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ape/man/axisPhylo.html'>axisPhylo</a></span>()</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/graphics/mtext.html'>mtext</a></span>(<span class='st'>"Time (million years ago)"</span>, <span class='kw'>side</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>line</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>at</span> <span class='kw'>=</span> (<span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/get.html'>get</a></span>(<span class='st'>"last_plot.phylo"</span>,
                <span class='kw'>envir</span> <span class='kw'>=</span> <span class='no'>.PlotPhyloEnv</span>)$<span class='no'>xx</span>) * <span class='fl'>0.5</span>))</div><div class='img'><img src='datelife_search-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='https://rdrr.io/pkg/ape/man/write.tree.html'>write.tree</a></span>(<span class='no'>ages.again</span>, <span class='kw'>file</span><span class='kw'>=</span><span class='st'>"some.bird.tree.again.txt"</span>) <span class='co'># saves phylo object in newick format</span>

<span class='co'># obtain mrca ages and target chronograms from all source chronograms</span>
<span class='co'># generate an html  output readable in any web browser:</span>
<span class='no'>ages.html</span> <span class='kw'>&lt;-</span> <span class='fu'>datelife_search</span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"Rhea americana"</span>, <span class='st'>"Pterocnemia pennata"</span>, <span class='st'>"Struthio camelus"</span>,
                <span class='st'>"Mus musculus"</span>), <span class='kw'>summary_format</span><span class='kw'>=</span><span class='st'>"html"</span>)</div><div class='output co'>#&gt; <span class='message'>Input taxa presence across source chronograms:</span></div><div class='output co'>#&gt; <span class='message'>                taxon chronograms</span>
#&gt; <span class='message'>1 Pterocnemia pennata         6/9</span>
#&gt; <span class='message'>2      Rhea americana         9/9</span>
#&gt; <span class='message'>3    Struthio camelus         9/9</span>
#&gt; <span class='message'>4        Mus musculus         1/9</span></div><div class='output co'>#&gt; <span class='message'>Input taxa completely absent from source chronograms:</span></div><div class='output co'>#&gt; <span class='message'>  taxon</span>
#&gt; <span class='message'>1  None</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/write.html'>write</a></span>(<span class='no'>ages.html</span>, <span class='kw'>file</span><span class='kw'>=</span><span class='st'>"some.bird.trees.html"</span>)
<span class='fu'><a href='https://rdrr.io/r/base/system.html'>system</a></span>(<span class='st'>"open some.bird.trees.html"</span>)</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian O'Meara, Luna L. Sanchez-Reyes, Jonathan Eastman, Tracy Heath, April Wright, Klaus Schliep, Scott Chamberlain, Peter Midford, Luke Harmon, Joseph Brown, Matt Pennell, Mike Alfaro.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


