<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Use genetic data from the Barcode of Life Database (BOLD) to reconstruct branch lengths on a tree. — make_bold_otol_tree • datelife</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Use genetic data from the Barcode of Life Database (BOLD) to reconstruct branch lengths on a tree. — make_bold_otol_tree"><meta property="og:description" content="make_bold_otol_tree takes taxon names from a tree topology or
a vector of names to search for genetic markers in the Barcode of Life Database
(BOLD), create an alignment, and reconstruct branch lengths on a tree topology
with Maximum Likelihood."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">datelife</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Getting_started_with_datelife.html">Getting started with `datelife`</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/phylotastic/datelife/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Use genetic data from the Barcode of Life Database (BOLD) to reconstruct branch lengths on a tree.</h1>
    <small class="dont-index">Source: <a href="https://github.com/phylotastic/datelife/blob/HEAD/R/bold_tree.R" class="external-link"><code>R/bold_tree.R</code></a></small>
    <div class="hidden name"><code>make_bold_otol_tree.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>make_bold_otol_tree</code> takes taxon names from a tree topology or
a vector of names to search for genetic markers in the Barcode of Life Database
(BOLD), create an alignment, and reconstruct branch lengths on a tree topology
with Maximum Likelihood.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">make_bold_otol_tree</span><span class="op">(</span>
  input <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Rhea americana"</span>, <span class="st">"Struthio camelus"</span>, <span class="st">"Gallus gallus"</span><span class="op">)</span>,
  marker <span class="op">=</span> <span class="st">"COI"</span>,
  otol_version <span class="op">=</span> <span class="st">"v3"</span>,
  chronogram <span class="op">=</span> <span class="cn">TRUE</span>,
  doML <span class="op">=</span> <span class="cn">FALSE</span>,
  aligner <span class="op">=</span> <span class="st">"muscle"</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>input</dt>
<dd><p>One of the following:</p><dl><dt>A character vector</dt>
<dd><p>With taxon names as a single comma separated strting or concatenated with <code><a href="https://rdrr.io/r/base/c.html" class="external-link">c()</a></code>.</p></dd>

<dt>A phylogenetic tree with taxon names as tip labels</dt>
<dd><p>As a <code>phylo</code> or <code>multiPhylo</code>
object, OR as a newick character string.</p></dd>

<dt>A <code>datelifeQuery</code> object</dt>
<dd><p>An output from <code><a href="make_datelife_query.html">make_datelife_query()</a></code>.</p></dd>


</dl></dd>
<dt>marker</dt>
<dd><p>A character vector indicating the gene from BOLD system to be
used for branch length estimation.</p></dd>
<dt>otol_version</dt>
<dd><p>Version of Open Tree of Life to use</p></dd>
<dt>chronogram</dt>
<dd><p>Default to <code>TRUE</code>, branch lengths returned are estimated with
<code><a href="https://rdrr.io/pkg/ape/man/chronoMPL.html" class="external-link">ape::chronoMPL()</a></code>. If <code>FALSE</code>, branch lengths returned are estimated with
<code><a href="https://rdrr.io/pkg/phangorn/man/parsimony.html" class="external-link">phangorn::acctran()</a></code> and represent relative substitution rates.</p></dd>
<dt>doML</dt>
<dd><p>Default to <code>FALSE</code>. If <code>TRUE</code>, it does ML branch length optimization
with <code><a href="https://rdrr.io/pkg/phangorn/man/pml.html" class="external-link">phangorn::optim.pml()</a></code>. Only relevant if chronogram = <code>TRUE</code>.</p></dd>
<dt>aligner</dt>
<dd><p>A character vector indicating whether to use MAFFT or MUSCLE
to align BOLD sequences. It is not case sensitive. Default to MUSCLE,
supported using the <a href="https://bioconductor.org/packages/release/bioc/html/msa.html" class="external-link">msa</a>
package from Bioconductor. Needs to be installed using <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">BiocManager::install()</a></code>.</p></dd>
<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="get_otol_synthetic_tree.html">get_otol_synthetic_tree</a></code></p><dl><dt><code>resolve</code></dt>
<dd><p>boolean; default to TRUE. Whether to resolve the tree at random or not.</p></dd>

    <dt><code>ott_ids</code></dt>
<dd><p>If not NULL, it takes this argument and ignores input. A
numeric vector of ott ids obtained with <code><a href="https://docs.ropensci.org/rotl/reference/taxonomy_taxon_info.html" class="external-link">rotl::taxonomy_taxon_info()</a></code> or
<code><a href="https://docs.ropensci.org/rotl/reference/tnrs_match_names.html" class="external-link">rotl::tnrs_match_names()</a></code> or <code><a href="tnrs_match.html">tnrs_match()</a></code></p></dd>

  
</dl></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A phylogeny with branch lengths poportional to relative substitution rate.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>If input is a phylo object or a newick string, it is used as backbone topology.
If input is a character vector of taxon names, an induced synthetic OpenTree
subtree is used as backbone.
If there are not enough sequences to return a tree with branch lengths, it returns
either the original input topology or the synthetic OpenTree subtree obtained
for the input taxon names.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian O’Meara, Jonathan Eastman, Tracy Heath, April Wright, Klaus Schliep, Scott Chamberlain, Peter Midford, Luke Harmon, Joseph Brown, Matt Pennell, Mike Alfaro, Luna L. Sanchez Reyes.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer></div>

  


  

  </body></html>

